turbopy/core.py

```diff
@@ -942,7 +937,13 @@ def initialize(self):
 
         This gets called once before the main simulation loop.
```

Comment:
We should probably add a note to the docstring here that this base class definition of `initialize` does the work of creating the output directory if it does not exist. And also note that this should probably be called using `super().initialize()` if a subclass is going to override this function.